core/dbt/adapters/base/column.py

```diff
@@ -52,8 +50,27 @@ def is_string(self) -> bool:
         return self.dtype.lower() in ['text', 'character varying', 'character',
                                       'varchar']
 
+    def is_number(self):
+        return any([self.is_integer(), self.is_numeric(), self.is_float()])
+
+    def is_float(self):
+        return self.dtype.lower() in [
+            # floats
+            'real', 'float4', 'float', 'double precision', 'float8'
+        ]
+
+    def is_integer(self) -> bool:
+        return self.dtype.lower() in [
+            # real types
+            'smallint', 'integer', 'bigint',
+            'smallserial', 'serial', 'bigserial',
+            # aliases
+            'int2', 'int4', 'int8',
+            'serial2', 'serial4', 'serial8',
+        ]
+
     def is_numeric(self) -> bool:
-        return self.dtype.lower() in ['numeric', 'number']
+        return self.dtype.lower() in ['numeric', 'decimal']
```

Comment:
This change is the only thing in this PR that I'm not sure about -- do you know why `number` was in here before? I tried looking through the blame but couldn't find the origin.... AFAICT `number` is not a valid type on pg/redshift/snowflake/bq.....?

Reply:
It's only valid on snowflake, and previously this method applied for Snowflake as well as postgres/redshift. That's why I removed it when I split snowflake out.